tests/main/interfaces-cifs-mount/task.yaml

```diff
@@ -0,0 +1,66 @@
+summary: Ensure that the cifs-mount interface works.
+
+details: |
+    The cifs-mount interface allows mounting and unmounting CIFS filesystems.
+
+prepare: |
+    snap install test-snapd-cifs-mount
+
+    # Create linux and samba users 
+    useradd -s /bin/true test-cifs
+    echo "test-cifs:test123#" | chpasswd
+    mkdir /home/test-cifs
+
+    echo test > test-pass
+    echo test >> test-pass
+    smbpasswd -a test-cifs -s < test-pass
+
+    # Create mount point
+    mkdir -p local_share
+    test-snapd-cifs-mount.sh -c "mkdir \$SNAP_DATA/local_mount"
+    chmod 755 local_share
+    touch local_share/test
+
+    # Configure samba
+    cp /etc/samba/smb.conf /etc/samba/smb.conf.back
+    { echo "[data]"; echo "path=$(pwd)/local_share"; echo "writable=yes"; echo "browsable=yes"; } >> /etc/samba/smb.conf
+
+    systemctl restart smbd.service
+
+restore: |
+    rm -f call.error test-pass
+    rm -rf local_share /var/snap/test-snapd-cifs-mount/current/local_mount
+
+    # Restore samba configuration
+    smbpasswd -x test-cifs || true
+    userdel test-cifs
+    cp /etc/samba/smb.conf.back /etc/samba/smb.conf || true
+    systemctl restart smbd.service
+
+execute: |
+    echo "The interface is not connected by default"
+    snap interfaces -i cifs-mount | MATCH -- '^- +test-snapd-cifs-mount:cifs-mount'
```

Comment:
Why the --?

Reply:
Not strictly needed here because `'^..` will be detected as part of the pattern and handled porperly. However, the pattern `MATCH '- +test..'` will fail because `'-` will be seen as a `-foo` switch. So it's either `MATCH -- <pattern>`. We could use `grep .. -e`  switch which makes grep expec `<pattern>` as the next argument.